src/cli/commands/joinNetwork.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { ArweaveSigner } from '@dha-team/arbundles/node';
+import prompts from 'prompts';
+
+import { WalletOptions } from '../options.js';
+import {
+  jwkToAddress,
+  requiredJwkFromOptions,
+  writeIOFromOptions,
+} from '../utils.js';
+
+export type JoinNetworkOptions = WalletOptions & {
+  quantity: number | undefined;
+  disableAutoStake: boolean;
+  disableDelegatedStaking: boolean;
+  minDelegatedStake: number | undefined;
+  delegateRewardShareRatio: number | undefined;
+  label: string | undefined;
+  note: string | undefined;
+  properties: string | undefined;
+  observer: string | undefined;
+  fqdn: string | undefined;
+  port: number | undefined;
+  protocol: string | undefined;
+  allowedDelegates: string[] | undefined;
+};
```

Comment:
most of the types live here - https://github.com/ar-io/ar-io-sdk/blob/main/src/types/io.ts#L302

Reply:
I think it actually makes it more confusing because we end up going like `type = WalletOptions & Partial<Pick<AoJoinNetworkParams, 'label' | 'note' | '...'>> & { observer, disableAutoStaking, }` rather than listing out all the possible incoming command options

Reply:
is there some wrapper we can create to avoid retyping everything to be compliant with the API? even if that means `Partial<StringAllTypes<JoinNetworkParams>` - worry about maintaining two sets of types for CLI support that just creates more overhead/slows down shipping

Reply:
fed363a

still like UX the `disable` style options, but the type turned out pretty nice:
![image](https://github.com/user-attachments/assets/a8ab0e68-be7c-4f1e-bd69-2b95c67cb604)

Reply:
fwiw StringAllNumberTypes could ingest the `Partial` logic, maybe just `ParseCLITypes`